staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -40,6 +40,7 @@ import (
 	listers "k8s.io/apiextensions-apiserver/pkg/client/listers/apiextensions/internalversion"
```

Comment:
missing disabling of server-side apply patch type for non-structural schemas, compare https://github.com/kubernetes/kubernetes/pull/79433/commits/437dd2084d60ea084e2b5ef398a522e996cc4c3d#diff-2f720a7e61a4ddfcfe9a9e353be8784bR333

Reply:
Schemaless does not exist anymore in v1. There is always a schema, even if it says "everything is raw json".

We have been very strict to restrict any new feature to structural CRDs, with defaulting even to v1. I don't see why server-side-apply is any different.

Reply:
One technical reason: go-openapi will go away. We will always go through our `*Structural` type. I don't want server-side apply code here to hold us back, forcing us to maintain legacy code paths, for a new feature in 1.16. That doesn't make sense.

Reply:
Maybe I lack understanding here. So you are saying the feature makes sense for users who consume a 3rdparty CRD that is old and uncontrollable by them. So these users use server-side-apply against those CRs and the 3rdparty controller does not, i.e. uses normal Update or Patch. Those users will still benefit from SSA? The old controller will not interfere with those users?

Reply:
Ok, then I see the point.

@jennybuckley we need tests for these cases then: 1) no schema 2) non-structural schema 3) structural schema. Not sure which of these we cover already.